receiver/dockerstatsreceiver/metric_helper.go

```diff
@@ -71,3 +76,54 @@ func calculateMemoryPercent(limit uint64, usedNoCache uint64) float64 {
 	}
 	return 0.0
 }
+
+// calculateCPULimit calculate the number of cpus assigned to a container.
+//
+// Calculation is based on 3 alternatives by the following order:
+// - nanocpus:   if set by i.e docker run -cpus=2
+// - cpusetCpus: if set by i.e docker run -docker run -cpuset-cpus="0,2"
+// - cpuquota:   if set by i.e docker run -cpu-quota=50000
+//
+// See https://docs.docker.com/config/containers/resource_constraints/#configure-the-default-cfs-scheduler for background.
+func calculateCPULimit(hostConfig *ctypes.HostConfig) (float64, error) {
+	var cpuLimit float64
+	var err error
+
+	switch {
+	case hostConfig.NanoCPUs > 0:
+		cpuLimit = float64(hostConfig.NanoCPUs) / 1e9
+	case hostConfig.CpusetCpus != "":
+		cpuLimit, err = parseCPUSet(hostConfig.CpusetCpus)
+		if err != nil {
+			return cpuLimit, err
+		}
+	case hostConfig.CPUQuota > 0:
+		period := hostConfig.CPUPeriod
+		if period == 0 {
+			period = 100000 // Default CFS Period
+		}
+		cpuLimit = float64(hostConfig.CPUQuota) / float64(period)
+	}
+	return cpuLimit, nil
+}
+
+// parseCPUSet helper function to decompose -cpuset-cpus value into number os cpus.
+func parseCPUSet(line string) (float64, error) {
+	var numCPUs uint64
+
+	lineSlice := strings.Split(line, ",")
+	for _, l := range lineSlice {
+		lineParts := strings.Split(l, "-")
+		if len(lineParts) == 2 {
+			p0, err0 := strconv.Atoi(lineParts[0])
+			p1, err1 := strconv.Atoi(lineParts[1])
+			if err0 != nil || err1 != nil {
+				return 0, fmt.Errorf("invalid cpusetCpus value")
+			}
```

Comment:
```suggestion
			p0, err0 := strconv.Atoi(lineParts[0])
			if err0 != nil {
				return 0, fmt.Errorf("invalid cpusetCpus value: %w", err0)
			}
			p1, err1 := strconv.Atoi(lineParts[1])
			if err1 != nil {
				return 0, fmt.Errorf("invalid cpusetCpus value: %w", err1)
			}
```
I was thinking to suggest wrapping the error, so that it is clear what exactly happened.